src/westpa/westext/weed/weed_driver.py

```diff
@@ -95,7 +95,15 @@ def prepare_new_iteration(self):
 
         with self.data_manager.lock:
             weed_global_group = self.data_manager.we_h5file.require_group('weed')
+            reweighting_history_dataset = weed_global_group.require_dataset(
+                'reweighting_history', (1,), maxshape=(None,), dtype=int
+            )
             last_reweighting = int(weed_global_group.attrs.get('last_reweighting', 0))
+            if last_reweighting > n_iter:
+                last_reweighting = n_iter - 1
+                reweighting_history = np.array(reweighting_history_dataset)
```

Comment:
I would just change this to `reweighting_history = reweighting_history_dataset[:]` Less of a performance hit.